// @ionic-native/firebase-crash/__ivy_ngcc__/ngx/index.js
import { __decorate, __extends } from "tslib";
import { Injectable } from '@angular/core';
import { IonicNativePlugin, cordova } from '@ionic-native/core';
import * as ɵngcc0 from '@angular/core';
var FirebaseCrash = /** @class */ (function (_super) {
    __extends(FirebaseCrash, _super);
    function FirebaseCrash() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    FirebaseCrash.prototype.log = function (message) { return cordova(this, "log", { "sync": true }, arguments); };
    FirebaseCrash.prototype.logError = function (message) { return cordova(this, "logError", { "sync": true }, arguments); };
    FirebaseCrash.prototype.setUserId = function (userId) { return cordova(this, "setUserId", { "sync": true }, arguments); };
    FirebaseCrash.pluginName = "FirebaseCrash";
    FirebaseCrash.plugin = "cordova-plugin-firebase-crash";
    FirebaseCrash.pluginRef = "cordova.plugins.firebase.crashlytics";
    FirebaseCrash.repo = "https://github.com/chemerisuk/cordova-plugin-firebase-crash";
    FirebaseCrash.platforms = ["Android", "iOS"];
FirebaseCrash.ɵfac = function FirebaseCrash_Factory(t) { return ɵFirebaseCrash_BaseFactory(t || FirebaseCrash); };
FirebaseCrash.ɵprov = ɵngcc0.ɵɵdefineInjectable({ token: FirebaseCrash, factory: function (t) { return FirebaseCrash.ɵfac(t); } });
var ɵFirebaseCrash_BaseFactory = ɵngcc0.ɵɵgetInheritedFactory(FirebaseCrash);
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(FirebaseCrash, [{
        type: Injectable
    }], null, null); })();
    return FirebaseCrash;
}(IonicNativePlugin));
export { FirebaseCrash };
